lib/pipe/bitbucket.js

```diff
@@ -0,0 +1,259 @@
+const debug = require('debug')('@testomatio/reporter:pipe:bitbucket');
+const { default: axios } = require('axios');
+const chalk = require('chalk');
+const humanizeDuration = require('humanize-duration');
+const merge = require('lodash.merge');
+const path = require('path');
+const { APP_PREFIX, testomatLogoURL } = require('../constants');
+const { ansiRegExp, isSameTest } = require('../utils/utils');
+const { statusEmoji, fullName } = require('../utils/pipe_utils');
+
+//! BITBUCKET_PAT environment variable is required for this functionality to work
+//! and your pipeline trigger should be a pull request
+
+/**
+ * @class BitbucketPipe
+ * @typedef {import('../../types').Pipe} Pipe
+ * @typedef {import('../../types').TestData} TestData
+ */
+class BitbucketPipe {
+  constructor(params, store = {}) {
+    this.isEnabled = false;
+    this.ENV = process.env;
+    this.store = store;
+    this.tests = [];
+    // Bitbucket PAT looks like bbpat-*****
+    this.token = params.BITBUCKET_PAT || process.env.BITBUCKET_PAT || this.ENV.BITBUCKET_PAT;
+    this.hiddenCommentData = `Testomat.io report: ${process.env.BITBUCKET_BRANCH || ''}`;
+
+    debug(
+      chalk.yellow('Bitbucket Pipe:'),
+      this.token ? 'TOKEN passed' : '*no token*',
+      `Project key: ${this.ENV.BITBUCKET_PROJECT_KEY}, Pull request ID: ${this.ENV.BITBUCKET_PR_ID}`,
+    );
+
+    if (!this.ENV.BITBUCKET_PROJECT_KEY || !this.ENV.BITBUCKET_PR_ID) {
+      debug(`CI pipeline should be run in a Pull Request to have the ability to add the report comment.`);
```

Comment:
```suggestion
```
As this pipeline won't be enabled for most of cases, we don't need this debug message

Reply:
Done